apps/web/src/app/_components/reviews/new-role-dialogue.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Button } from "@cooper/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@cooper/ui/dialog";
+import { Form, FormControl, FormField, FormItem } from "@cooper/ui/form";
+import { useToast } from "@cooper/ui/hooks/use-toast";
+import { Input } from "@cooper/ui/input";
+import { Label } from "@cooper/ui/label";
+import { Textarea } from "@cooper/ui/textarea";
+
+import { api } from "~/trpc/react";
+import { auth } from "@cooper/auth";
+
+const roleSchema = z.object({
+  title: z
+    .string({ required_error: "You need to enter a role title." })
+    .min(10, {
```

Comment:
Lowkey maybe we should add .max to these as well since if we are letting anyone enter things, they might try to be malicious or smth and leave like 100000 characters 💀